src/patterns/ec2-app/framework.ts

```diff
@@ -4,11 +4,31 @@ The plan is to sprinkle some framework specific tooling into them.
 For example, a Play app should come with the infrastructure for https://github.com/guardian/play-secret-rotation.
  */
 
+import { AccessScope } from "../../constants";
 import type { GuStack } from "../../constructs/core";
+import type { GuDomainName } from "../../types";
 import type { GuEc2AppProps } from "./base";
 import { GuEc2App } from "./base";
 
-type GuEc2FrameworkAppProps = Omit<GuEc2AppProps, "applicationPort">;
+type GuEc2FrameworkAppProps = Omit<GuEc2AppProps, "applicationPort"> & { certificateProps: GuDomainName };
+
+type GuEc2WorkerProps = Omit<GuEc2AppProps, "applicationPort" | "access">;
+
+/**
+ * Creates an instance of [[`GuEc2App`]], with an application port of 9000 and no access to the load balancer.
+ * This is useful for applications that only have an LB for health check purposes
```

Comment:
I agree that in many cases Lambda is the way to go.  But this service is http long polling an external API, so needs to be running continually. I believe this will be cheaper as an instance rather than a lambda.  I suspect that services like big loader are also processing enough data that it is also more cost effective to to run a worker.

Perhaps we just need a rule of thumb of when it is more cost effective to use this pattern. And to create a nice pattern to make the Lambda path appealing.

Reply:
If the concern is about docs rather than the pattern itself, can we merge this as is and follow up with docs? This is currently blocking another piece of work.

Reply:
> I agree that in many cases Lambda is the way to go. But this service is http long polling an external API, so needs to be running continually.

Thanks for explaining, that makes sense 👍 

> can we merge this as is and follow up with docs?

Yes, I don't consider this question about docs to be blocking (sorry I should have made that clearer in my initial comment). I will let someone from the @guardian/devx-operations team provide the official review (I'm not in that team any more), but 👍  from my perspective.